infra/profile.py

```diff
@@ -69,16 +72,41 @@ def setup_benchmark(name):
   profile_dir = benchmark_profile_dir(name)
   os.mkdir(profile_dir)
 
+
+# runs optimization on the benchmark, a dictionary with
+# a name and treatment (and index for printing)
+# returns a dictionary with the path to the optimized binary,
+# eggcc compile time, and llvm compile time
 def optimize(benchmark):
   print(f'[{benchmark.index}/{benchmark.total}] Optimizing {benchmark.name} with {benchmark.treatment}')
   profile_dir = benchmark_profile_dir(benchmark.name)
-  cmd = f'cargo run --release {benchmark.path} --add-timing {get_eggcc_options(benchmark.treatment, benchmark.name)} -o {profile_dir}/{benchmark.treatment}'
-  print(f'Running: {cmd}', flush=True)
-  start = time.time()
-  process = subprocess.run(cmd, shell=True)
+  optimized_bril_file = f'{profile_dir}/{benchmark.treatment}-optimized.bril'
+
+  # get the commands we need to run
+  (eggcc_run_mode, llvm_args) = get_eggcc_options(benchmark)
+  llvm_out_dir = f"{DATA_DIR}/llvm/{benchmark.name}/{benchmark.treatment}"
+
+  cmd1 = f'{EGGCC_BINARY} {benchmark.path} --run-mode {eggcc_run_mode}'
+  cmd2 = f'{EGGCC_BINARY} {optimized_bril_file} --add-timing {llvm_args} -o {profile_dir}/{benchmark.treatment} --llvm-output-dir {llvm_out_dir}'
+
+  print(f'Running: {cmd1}', flush=True)
```

Comment:
probably delete?

Reply:
I find this useful to see what it's really doing, but it's personal preference